ffi/Cargo.toml

```diff
@@ -38,7 +38,13 @@ trybuild = "1.0"
 
 [features]
 default = ["default-engine"]
-default-engine = ["delta_kernel/default-engine", "delta_kernel/arrow"]
+default-engine = ["delta_kernel/default-engine", "default-engine-base"]
+default-engine-rustls = ["delta_kernel/default-engine-rustls", "default-engine-base"]
+
+# this is an 'internal' feature flag which has all the shared bits from default-engine and
+# default-engine-rustls
```

Comment:
nit: can we add a comment that the check in kernel/lib.rs that you've enabled one of `default-engine` or `default-engine-rustls` will ensure that we don't _just_ enable `default-engine-base` here too.